flexmeasures/data/models/user.py

```diff
@@ -289,6 +289,73 @@ def has_role(self, role: str | Role) -> bool:
             return role in self.flexmeasures_roles
 
 
+class AuditLog(db.Model, AuthModelMixin):
+    """
+    Model for storing actions that happen to user and tenant accounts
+    E.g user creation, password reset etc.
+    """
+
+    __tablename__ = "audit_log"
+    id = Column(Integer, primary_key=True)
+    event_datetime = Column(DateTime())
+    event = Column(String(255))
+    active_user_id = Column("active_user_id", Integer(), ForeignKey("fm_user.id"))
+    affected_user_id = Column("affected_user_id", Integer(), ForeignKey("fm_user.id"))
+    affected_account_id = Column(
+        "affected_account_id", Integer(), ForeignKey("account.id")
+    )
+
+    @classmethod
+    def user_acl(cls, user: User):
+        """
+        User can see his own audit logs.
+        Account-admin users can see audit logs for all users of their account.
+        Admins / admin-readers can see audit logs for all users.
+        Consultant users can see the audit log of all users in the client accounts.
+        """
+
+        class AuditLogAccess(AuthModelMixin):
+            def __init__(self, user):
+                if user:
```

Comment:
What if no `user` is passed? I guess then we have failures when the `__acl__` function is called.